### ISSUE TITLE
Data import from file not working in mobile android app

### ISSUE THREAD

**Issue by captzilla** [johannesjo/super-productivity#746]
### Your Environment

* Version used:  google play super productivity app 6.1.2
* Operating System and version:  protonAOSP android 11 
* Desktop Environment:  
* Browser Name and version: 

### Expected Behavior
 file browser should open to locate file to upload

### Current Behavior

nothing happens

### Steps to Reproduce (for bugs)

1.
2.
3.
4.

**Comment by johannesjo** [comment]
Hey there! Thanks for opening this up (and merry Christmas btw! :)). The app does not support file uploads as there is no data saved to an external server. Attachments are merely linked either from your hard drive or any web url.

**Comment by captzilla** [comment]
happy holidays as well. I meant to say it doesn't let me attach the file from my hard drive. Nothing happens when I touch the "import from file" button

**Comment by johannesjo** [comment]
Alright! I can confirm that this isn't working for the android app (it is however for the PWA).

**Comment by ffcruz85** [comment]
@johannesjo has this been completely removed? I don´t see it on the android menu. I had never used file attachments before, but just seeing it on the desktop now made me think it would be super nice to have it on mobile. The use case i had in mind was for decluttering my desk actually - I have random bills or pieces of paper I have to perform a task for, so create a task, snap photo of the related paper (like it has the URL/PIN to pay a bill, for example) and attach to task, and throw away piece of paper. So in the use case I made, ideal UI is the kind you see in say the messages app, where you click to attach image and it offers to pick from gallery or snap one right then - but this is image specific, one can always take a photo and pick from files :) I can imagine it being useful for some stuff on the go (see flyer on the street for some event - create task buy tickets and attach photo of the flyer, that kind of thing).
I guess one issue here is that syncing wouldn't be a possibility, right? I mean, it could sync to google drive I guess, but that's further development on your end

**Comment by johannesjo** [comment]
> I guess one issue here is that syncing wouldn't be a possibility, right? I mean, it could sync to google drive I guess, but that's further development on your end

Exactly. I am all for adding support for something like this, but I likely won't find the time to work on this anytime soon.

**Comment by M007RA** [comment]
Hi! Please assign this issue to me. I can work on it.

**Comment by lrq3000** [comment]
@johannesjo @M007RA I can't code in Typescript, so I can't work on the webapp, but I can work on the Android side. If you can make the button callback a Kotlin function, similarly to what was done for the file sync feature, I can take care of the rest (opening a file chooser dialog, getting permissions, opening file for read/write, etc).

**Comment by johannesjo** [comment]
That's great to hear! I can do this  @lrq3000 ! I try to look into it next week!

**Comment by lrq3000** [comment]
Ok awesome! Just ping me when the typescript side is ready and I'll work on it! :-)

14 janv. 2023 21:16:33 Johannes Millan ***@***.***>:

**Comment by ffcruz85** [comment]
That's great news! I wonder if, since this will provide the code needed to work with files (if I understand correctly), it could be also used as the solution to provide file syncing in Android? When the Google drive issue came up, at first I tried WebDAV and failed, but my immediate thought was I already have some cloud file syncing solution (Synology NAS Drive client in both my PC and android), so why not use the sync to file option and let the other software handle the sync between the devices, and super prod will just read the synced file. I set it up on the PC, but then saw that option wasn't available in Android. If syncing to the file system on Android was possible, people could use any file/cloud sync solution they already have, without super prod having to deal with the APIs of specific cloud services and their changes/updates. Am I reading this wrong or missing something @johannesjo ?

**Comment by lrq3000** [comment]
@ffcruz85 It's the other way around, I made a PR to implement file syncing on Android already, and later saw this issue about importing files, which can be solved similarly if some adjustments in the typescript webapp are made :-)

**Comment by lrq3000** [comment]
@ffcruz85 Forgot to link to PR for filesync on Android, see: https://github.com/johannesjo/super-productivity-android/pull/33

**Comment by ffcruz85** [comment]
@lrq3000 oh man, talk about missing the context! You see, my productivity was down, because super productivity wasn't syncing 😂

Ok, thanks for your work implementing this, I wish I could contribute here but both typescript and mobile development are out of my lane. But, from the PR, it seems I could help test? Just use the APK you linked?

**Comment by lrq3000** [comment]
@ffcruz85 No worries, it's normal, this was all discussed in another repository entirely that was not linked in the current discussion :-)

Yes please if you can test the debug APK and report whether it works fine for you, that would be awesome! Just please make sure to make a backup of your file beforehand on your desktop computer, in case something bad happens (such as the android app mangling the file)!

**Comment by ffcruz85** [comment]
Sounds good, checking it out now. Mostly trying to figure out where my Synology drive client on Android keeps offline files at the moment, but if for some reason it doesn't (Google searches indicating it might not in that simple way, even when offline access requested, sigh), I will try another sync solution just to help out with testing. PS.: Do you have any suggestions for a good android file manager app where I can properly see the file system, unlike my manufacturer's "user friendly" idea of a file manager app

**Comment by ffcruz85** [comment]
For example, my file managers response to whatever the default folder supprod was trying to open when I clicked select location in sync settings:![Screenshot_2023-01-14-22-02-00-16_5734e8eb49b4234b62f913f831715b0f.jpg](https://user-images.githubusercontent.com/2352828/212522353-0c70af8f-0653-400b-ab08-95afea0a4349.jpg)

**Comment by lrq3000** [comment]
@ffcruz85 that's normal, it's because you are at the root and you can't use this folder with new Android versions. Did you try to create a new folder here?

**Comment by lrq3000** [comment]
@ffcruz85 Also about what file manager I suggest, there is Total Commander free but not open-source app, or Ghost Commander if you want an open-source equivalent but with less features.

**Comment by ffcruz85** [comment]
Sorry, I get it and I was speaking more to my specific sync situation which is not helpful here. Ok,  I created a folder at the default root location it offered me, tapped use that on my file manager, asked me for permission, tapped allow and it sent me back to super productivity. Let me figure out my own syncing situation which is not as transparent as I thought and I will get back to you

**Comment by lrq3000** [comment]
Unfortunately, after some experimentations, it seems Android's WebView is quite finicky, I lack the expertise to fully implement a file upload feature on Android. So I must withdraw from implementing this feature. Nevertheless, file synchronization should hopefully be merged soon-ish (it is working for me, but there are some edge case bugs to be polished before merging in the main branch).

**Comment by ffcruz85** [comment]
@lrq3000 sorry I dropped out on you - my semester started and it got crazy busy quickly (teaching), before I even figured out my own file sync situation - which tbh, wasn't actually needed to help with testing in the first place. I see that the PR merged, so great! Thanks for your contribution to freeing us from cloud services :)

**Comment by lrq3000** [comment]
@ffcruz85 no worries, thank you for your feedback! Yes now the filesync feature has beee merged. I unfortunately still cannot work on file attachments because I'd need to experiment on both the webapp and the android app simultaneously, but I can't make my local webapp work inside the Android Studio emulator (see https://github.com/johannesjo/super-productivity-android/issues/34). So someone else will have to try, hopefully my work on file sync can serve as an inspiration on how to handle file access on Android.

**Comment by fraschm1998** [comment]
Any update on this?

**Comment by lrq3000** [comment]
@fraschm1998 well, modern Android file permission management is now merged in. File syncing works for Android. File attachment is not implemented because I don't have any idea how to do that, it seems that major Android WebView and async wizardry is required on top of typescript changes, so I will leave that to someone else with more experience on this than me.

**Comment by johannesjo** [comment]
> /edit: Ah if you are asking when will the new Android app with filesync be published, I have no idea. It should be publishable to the Google Play Store without any hitch now since the permissions are limited, but I guess Johannes the main dev wants to finish a few more bugfixes or features before publishing it officially.

Hey hey! Just a quick update: The new release is still being under evaluation by the google play store team. I hope they finish soon! :)

**Comment by lrq3000** [comment]
Oh sorry, I assumed you would have published a release and/or published to f-droid in parallel. I hope it will be validated soon, I can't see why they would refuse this time, fingers crossed!

6 mars 2023 16:08:01 Johannes Millan ***@***.***>:

**Comment by johannesjo** [comment]
I forgot that I need to add a new tag on the super-productivity-android for f-droid to be updated. I did now. Hopefully it'll work 🤞
